tests/forms_tests/tests/test_forms.py

```diff
@@ -1416,18 +1416,18 @@ class Person(Form):
             """<tr><th>First name:</th><td><input type="text" name="first_name" required></td></tr>
 <tr><th>Last name:</th><td><input type="text" name="last_name" required></td></tr>
 <tr><th>Birthday:</th>
-<td><input type="text" name="birthday" required><input type="hidden" name="hidden_text"></td></tr>"""
+<td><input type="text" name="birthday" required></td></tr><input type="hidden" name="hidden_text">"""
         )
         self.assertHTMLEqual(
             p.as_ul(),
             """<li>First name: <input type="text" name="first_name" required></li>
 <li>Last name: <input type="text" name="last_name" required></li>
-<li>Birthday: <input type="text" name="birthday" required><input type="hidden" name="hidden_text"></li>"""
+<li>Birthday: <input type="text" name="birthday" required></li><input type="hidden" name="hidden_text">"""
```

Comment:
I think it would be an easier change if we could maintain the current rendering. It doesn't seem that complex. A small adjustment to the `ul.html` template, for example: 

```
    {% if forloop.last %}
      {% for field in hidden_fields %}{{ field }}{% endfor %}
    {% endif %}
```

(Before the closing the `<li>`) 

What do you think?

Reply:
@carltongibson sadly it is rather complex. See my first commit for comparison. I did jump though the hoops of keeping the order, but after being done, I realized the template get too difficult to read, so I opted for changing the tests. I did keep both commits, so you can see what I mean.

Reply:
I think complexity in the templates is just a bullet we have to bite here.

Beyond whitespace, I don't think we can change the outputted HTML — not the DOM structrue. There will be years worth of (particularly) JavaScript targeting hidden form elements by CSS selector. We can't break that. 

Please can I ask you to maintain the previous behavior? (As I say, non-DOM affecting whitespace differences are OK.)